src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java

```diff
@@ -0,0 +1,204 @@
+                    preparedStatement.setInt(3, offset);
+                },
+                new TestScenarioMapper()
+        );
+    }
+
+    @Override
+    public TestScenario createTestScenario(TestScenario testScenario) {
+        jdbcTemplate.update(
+                CREATE_TEST_SCENARIO,
+                testScenario.getTitle(),
+                testScenario.getDescription(),
+                testScenario.getStatus().toString(),
+                testScenario.getIdTestCase()
+        );
+        return this.findByTitle(testScenario.getTitle());
+    }
+
+    @Override
+    public TestScenario editTestScenario(TestScenario testScenario, int id) {
+        jdbcTemplate.update(
+                EDIT_TEST_SCENARIO,
+                testScenario.getTitle(),
+                testScenario.getDescription(),
+                testScenario.getStatus().toString(),
+                testScenario.getIdTestCase(),
+                id
+        );
+        return findByTitle(testScenario.getTitle());
+    }
+
+    @Override
+    public void deleteTestScenario(int id) {
+        jdbcTemplate.update(DELETE_TEST_SCENARIO, id);
+    }
+
+    @Override
+    public int getTotalElements(int idTestCase, String title) {
+        List<Integer> count = jdbcTemplate.query(
+                GET_TOTAL_ELEMENTS_FROM_SEARCH,
+                preparedStatement -> {
+                    preparedStatement.setInt(1, idTestCase);
+                    preparedStatement.setString(2, title);
+                },
+                new CountTestScenarioRowMapper());
+        if (count.size() == 1) return count.get(0);
+        else return 0;
+    }
+
+    @Override
+    public int getTestScenariosFromTestCaseAmount(int idTestCase) {
+        List<Integer> count = jdbcTemplate.query(
+                GET_SCENARIOS_FROM_TEST_CASE_AMOUNT,
+                preparedStatement -> {
+                    preparedStatement.setInt(1, idTestCase);
+                },
+                new CountTestScenarioRowMapper());
+        if (count.size() == 1) return count.get(0);
+        else return 0;
+    }
+
+    @Override
+    public TestScenario findByTitle(String title) {
+        List<TestScenario> testScenarios = jdbcTemplate.query(
+                FIND_BY_TITLE,
+                preparedStatement -> preparedStatement.setString(1, title),
+                new TestScenarioMapper()
+        );
+        if (testScenarios.size() == 1) {
+            return testScenarios.get(0);
+        } else return null;
+    }
+
+    @Override
+    public List<TestScenario> getAllPaginated(int limit, int offset) {
+        return jdbcTemplate.query(
+                GET_ALL_TEST_SCENARIOS,
+                preparedStatement -> {
+                    preparedStatement.setInt(1, limit);
+                    preparedStatement.setInt(2, offset);
+                },
+                new TestScenarioMapper()
+        );
+    }
+
+    @Override
+    public int getAmountAllElements() {
```

Comment:
You don't need to implement this method because it is already implemented in AbstractDaoImpl. You can call it using AbstractDao.getAmountAllElements().